services/net/image/ImageAction.cs

```diff
@@ -100,15 +100,12 @@ public ImageAction(IApiService api, IOptions<ImageOptions> options, ILogger<Imag
                 {
                     // If another process has it in progress only attempt to do an import if it's
                     // more than an 5 minutes old. Assumption is that it is stuck.
-                    await this.UpdateContentReferenceAsync(reference, WorkflowStatus.InProgress);
+                    reference = await UpdateContentReferenceAsync(reference, WorkflowStatus.InProgress);
                 }
-                else reference = null;
+                else continue;
 
-                if (reference != null)
-                {
-                    await CopyImageAsync(client, manager.Ingest, remotePath.CombineWith(file.Name));
-                    await this.ContentReceivedAsync(manager, reference, CreateSourceContent(manager.Ingest, reference));
-                }
+                await CopyImageAsync(client, manager.Ingest, remotePath.CombineWith(file.Name));
+                if (reference != null) await ContentReceivedAsync(manager, reference, CreateSourceContent(manager.Ingest, reference));
```

Comment:
I think I see what you mentioned in another comment.  If `CopyImageAsync` takes a while it means the `reference` is old.  You will have to make a request for the `reference` after the `CopyImageAsync` and then handle null.